src/Microsoft.Restier.WebApi/Query/ODataCountOption.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Restier.WebApi.Query
+{
+    internal class ODataCountOption
+    {
+        /// <summary>
+        /// Gets or sets a value indicating whether the total
+        /// number of items should be retrieved when the
+        /// result has been filtered using paging operators.
+        /// </summary>
+        /// <remarks>
+        /// Setting this to <c>true</c> may have a performance impact as
+        /// the data provider may need to execute two independent queries.
+        /// </remarks>
+        public bool IncludeTotalCount { get; set; }
+
+        public Action<long> SetTotalCount { get; set; }
+    }
+}
```

Comment:
Since ApiContext now is also a scoped service, so attaching something to it basically has the same effect as registering a scoped service (like this ODataCountOption class).
And actually I'm planning to start a discussion about removing PropertyBag. As we have DI now, I'd discourage using any generic dictionary as data store, we'd use dedicated services instead. Of course eventually you guys will make the decision, but for now let's stop adding it to ApiContext property bag.
So let me summarize the design options here, actually we just need a decision on the mindset.
- Option1: We add cross-cutting feature services, that each serves a specific feature (like getting total count), but not bound to a specific platform or publisher (like OData/WebApi). This way we may result in multiple services and can choose among them to register.
- Option2: We add services that bound to specific platform and implement all features together, results in less services that can only work on the specific platform or publisher.

Since now we only have 1 such requirement, these 2 options are kind of indistinguishable, but later the difference will be more and more prominent.
So you'd like to go option 2 right? If so I'll rename classes accordingly and assume they're bound to OData/WebApi publisher.